Plutarch/Maybe.hs

```diff
@@ -192,3 +202,53 @@ pmapMaybe = phoistAcyclic $
   plam $ \f mv -> pmatch mv $ \case
     PJust v -> pjust # (f # v)
     PNothing -> pnothing
+
+-- | @since WIP
+data PMaybeSoP (a :: S -> Type) (s :: S)
```

Comment:
We can just change `PMaybe` to use SOP instead of having multiple types

Reply:
I think we can do it in separate PR, confirm that SoP improves PRational and probably drop Scott encoding altogether